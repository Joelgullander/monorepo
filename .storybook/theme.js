import { create } from '@storybook/theming';

const theme = create({
    base: 'light',
  
    colorPrimary: '#D52B1E',
    colorSecondary: '#D52B1E',
  
    // UI
    appBg: 'white',
    appContentBg: '#f9f9f9',
    appBorderColor: '#f000',
    appBorderRadius: 0,
  
    // Typography
    fontBase: '"Roboto", sans-serif',
    fontCode: 'monospace',
  
    // Text colors
    textColor: '#212121',
    textInverseColor: 'rgba(255,255,255,0.9)',
  
    // Toolbar default and active colors
    barTextColor: 'white',
    barSelectedColor: '#D52B1E',
    barBg: '#000',
  
    // Form colors
    inputBg: 'white',
    inputBorder: 'silver',
    inputTextColor: 'black',
    inputBorderRadius: 4,
  
    brandTitle: 'Hiab components',
    brandUrl: 'https://hiab.com',
    brandImage: 'https://www.hiab.com/Content/images/logotypes/hiab2x2.png',
  });

  export default theme;