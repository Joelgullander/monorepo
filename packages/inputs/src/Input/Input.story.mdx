import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { action } from '@storybook/addon-actions';

import {Input} from './Input';

<Meta title='Components|Inputs' component={Input} decorators={[withKnobs]} />

# Input

<Preview>
    <Story name='Input'>
        <Input 
            placeholder="Placeholder..." name="test" label="Label" 
            required={boolean("Required", true)} error={boolean("Error", true)} errorMessage="Something went wrong..." 
        />
    </Story>
</Preview>

<Preview>
    <Story name='Input Without Label'>
        <Input placeholder="Placeholder..." name="test" error={true} errorMessage="Something went wrong..." />
    </Story>
</Preview>


<Preview>
    <Story name='Multiple Inputs'>
        <>
            <Input placeholder="Placeholder..." name="firstName" label="First Name" />
            <Input placeholder="Placeholder..." name="lastName" label="Last Name" />
        </>
    </Story>
</Preview>

## Props

<Props of={Input} />
