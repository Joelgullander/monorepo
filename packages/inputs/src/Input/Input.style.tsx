import styled from 'styled-components';
import { HiabTheme } from '@hiab/theme';

interface IStyledButton {
    theme: HiabTheme,
    error?: boolean,
}

export const Input = styled.input`
    color: ${(props: IStyledButton) => props.theme.colors.mako};
    display: inline-block;
    padding: 10px 12px;
    border: 1px solid;
    border-color: #D6D6D6;
    font-size: 14px;
    border-bottom: ${(props: IStyledButton) => props.error ? `2px solid ${props.theme.colors.thunderbird}` : ''};
    box-sizing: border-box;

    &::placeholder {
        color: ${(props: IStyledButton) => props.theme.colors.osloGray};
    }

    &:focus {
        outline: none;
    }
`;

export const Wrapper = styled.div`
    display: inline-block;
    
    + div {
        margin-left: 10px;
    }

`

export const Label = styled.label<any>`
    display: flex;
    font-weight: bold;
    font-size: 14px;
    margin-bottom: 5px;
`;

export const Required = styled.div`
    color: ${(props: IStyledButton) => props.theme.colors.thunderbird};
    margin-left: 5px;
`;

export const ErrorMessage = styled.small`
    display: block;
    color: ${(props: IStyledButton) => props.theme.colors.thunderbird};
    font-size: 12px;
    margin-top: 5px;
    text-align: left;
`;