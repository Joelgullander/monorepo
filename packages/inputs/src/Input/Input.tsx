import React from 'react';
import * as style from './Input.style';

interface IInput {
  placeholder?: string,
  name: string,
  label?: string,
  required?: boolean,
  errorMessage?: string,
  error?: boolean,
  onChange?: any,
}

interface ILabel {
  name: string,
  label: string,
  required?: boolean,
}

const Label: React.FC<ILabel> = ({ name, label, required }) => (
  <style.Label htmlFor={name}>{label} {required && <style.Required>*</style.Required>}</style.Label>
);

export const Input: React.FC<IInput> = ({ placeholder, name, label, required, error, errorMessage, onChange }) => {
  // Do something with variant

  return (
    <style.Wrapper>
      {label && (
        <Label name={name} label={label} required={required} />)}

      <style.Input
        type="text"
        name={name}
        placeholder={placeholder}
        error={error}
        onChange={onChange}
      />

      {(error && errorMessage) && (
        <style.ErrorMessage>{errorMessage}</style.ErrorMessage>)}
    </style.Wrapper>
  );
};