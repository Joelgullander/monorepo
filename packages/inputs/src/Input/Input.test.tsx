import * as React from 'react';

import { shallow, mount } from 'enzyme';
import { Input } from './Input';
import { ThemeProvider } from '@hiab/theme';

const ThemeProviderWrapper : React.FC = ({ children }) => (
  <ThemeProvider>
    { children }
  </ThemeProvider>
);

export const shallowWithTheme = (tree: React.ReactElement) => shallow(tree, {
  wrappingComponent: ThemeProviderWrapper
});

export const mountWithTheme = (tree: React.ReactElement) => mount(tree, {
  wrappingComponent: ThemeProviderWrapper
});


describe('<Input />', () => {
  test('renders without exploding', () => {
    const wrapper = shallowWithTheme(<Input name="phone"/>)
    expect(wrapper.exists()).toBe(true);
  });

  test('snapshot', () => {
    const wrapper = shallowWithTheme(<Input name="phone" />);

    expect(wrapper).toMatchSnapshot();
  });

  test('prop name is passing', () => {
    const wrapper = mountWithTheme(<Input name="phone" />);
    
    expect(wrapper.props().name).toEqual("phone");
  });

  test('label is rendering', () => {
    const wrapper = mountWithTheme(
        <Input placeholder="Placeholder..." name="test" label="Label" required />
    );

    expect(wrapper.props().label).toEqual('Label');
  });
});
