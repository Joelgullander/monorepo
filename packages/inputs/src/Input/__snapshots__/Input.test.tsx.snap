// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot 1`] = `
<styled.div>
  <styled.input
    name="phone"
    type="text"
  />
</styled.div>
`;
